plugins/woocommerce/client/blocks/assets/js/shared/redux-stores/canvas-mode/index.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { createReduxStore, register, select, dispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { ACTION_SET_CANVAS_MODE, STORE_NAME } from './constants';
+import type { CanvasModeState } from './types';
+
+type StoreState = CanvasModeState;
+
+type Actions = {
+	type: typeof ACTION_SET_CANVAS_MODE;
+	isEditMode: boolean;
+};
+
+const DEFAULT_STATE: StoreState = {
+	isEditMode: false,
+};
+
+const actions = {
+	setCanvasMode( isEditMode: boolean ) {
+		return {
+			type: ACTION_SET_CANVAS_MODE,
+			isEditMode,
+		};
+	},
+};
+
+const selectors = {
+	isEditMode() {
+		// Always check the current URL for the most up-to-date state.
+		try {
+			const url = new URL( window.location.href );
+			const canvas = url.searchParams.get( 'canvas' );
+			return canvas === 'edit';
+		} catch ( error ) {
+			// If URL parsing fails, return default state.
+			return false;
+		}
+	},
+};
+
+if ( typeof window !== 'undefined' ) {
+	const handleUrlChange = async () => {
+		try {
+			const url = new URL( window.location.href );
+			const canvas = url.searchParams.get( 'canvas' );
+			const isEditMode = canvas === 'edit';
+
+			// Update store state to trigger any subscribed components.
+			const storeDispatch = dispatch( STORE_NAME ) as {
+				setCanvasMode: ( isEditMode: boolean ) => void;
+			};
```

Comment:
We can avoid defining the type manually using:

```suggestion
			dispatch(store).setCanvasMode(isEditMode)
```